### DISCUSSION TITLE
Replace original tessdata components before training

### DISCUSSION THREAD

**Discussion by Blu3train** [buliasz/tesstrain-windows-gui#26] [Ideas]
After some research I learned how to replace the component files of the original tessdata model to be trained.

It might be interesting to add the two buttons to unpack and the one for repack to the program to make the operation simple for novice users.
Actually there are other commands, like the one to replace a specific file without having to unpack first, but having the two basic operations mentioned above are enough to be able to do any replacement.

Everything is done with the [COMBINE_TESSDATA](https://github.com/tesseract-ocr/tesseract/blob/main/doc/combine_tessdata.1.asc) program.

Example:
For the following examples I have created the data/tessdata_unpacked folder

Command to unpack eng.traineddata:
`combine_tessdata -u data/tessdata/eng.traineddata data/tessdata_unpacked/eng`

...replace or edit the desired files, for example eng.config

Command to repack eng.traineddata:
`combine_tessdata data/tessdata_unpacked/eng`

**Comment by buliasz** [comment]
I'm not sure about this one. Adding unpack/repack buttons would need a lot of explanation, it opens a lot of possible issues when someone modifies or deletes wrong files etc. If user manages to do it by himself manually, they already know that they're responsible for what they will do there.

What I think we could do though is to allow specific values to be modified and then unpack/modify/repack automatically. Are there any specific options you have on your mind?

Another option is to show e.g. .config file and allow editing it within the GUI. Then once user has finished save and repack it automatically. Such simple editor is reasonably easy to do in AutoHotKey although I'm not sure if it is worth effort?

Even though this GUI is already quite advanced in possibilities I'm not sure if making it even more "for everything" is a good idea. Unless I'm wrong here, please convince me I'm wrong.

**Reply by Blu3train** [reply]
Yes, maybe it's better to do it manually. I changed the config file once to update it to the new published version. While the other thing I did was edit the "jpn.lstm-unicharset" file to add 3 special characters that I often encountered in the Japanese text I am training on (I am doing font training of some VNs and these 3 special characters ☆, ♪ and ♡ often appear).
